maven-surefire-common/src/test/java/org/apache/maven/plugin/surefire/SurefirePropertiesTest.java

```diff
@@ -70,9 +70,35 @@ public void testConstructWithOther()
         src.setProperty( "b", "2" );
         SurefireProperties orderedProperties = new SurefireProperties( src );
         // Cannot make assumptions about insertion order
-        assertEquals( 2, orderedProperties.size() );
-
+        int expectedCount = 0;
+        // keys() uses the items property, more reliable to test than size(),
+        // which is based on the Properties class
+        // see https://issues.apache.org/jira/browse/SUREFIRE-1445
+        Enumeration<Object> keys = orderedProperties.keys();
+        while ( keys.hasMoreElements() ) {
+            keys.nextElement();
+            expectedCount++;
+        }
+        assertEquals( 2, expectedCount );
```

Comment:
We missed one method I mentioned before. Can you add an assert that method `getStringKeySet()` returns same elements like it is in `keys()`. Maybe this will help:
`java.util.Collections#list(Enumeration): List`
We can do it in both tests.

Reply:
Got it, done.